docs/source/tutorials_source/platform/tutorial_active_learning_detectron2.py

```diff
@@ -0,0 +1,340 @@
+#
+# The length of the `unlabeled_set` should match the number of uploaded
+# images
+print(len(al_agent.unlabeled_set))
+
+# %%
+# Create our Detectron2 model
+# ----------------------------
+#
+# Next, we create a detectron2 config and a detectron2 `DefaultPredictor` to 
+# run predictions on the new images.
+# 
+# - We use a pre-trained Faster R-CNN with a ResNet-50 backbone
+# - We use a MS COCO pre-trained model from detectron2
+
+cfg = get_cfg()
+# add project-specific config (e.g., TensorMask) here if you're not running a model in detectron2's core library
+###cfg.merge_from_file(model_zoo.get_config_file("COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml"))
+cfg.merge_from_file(model_zoo.get_config_file("COCO-Detection/faster_rcnn_R_50_FPN_3x.yaml"))
+cfg.MODEL.ROI_HEADS.SCORE_THRESH_TEST = 0.5  # set threshold for this model
+# Find a model from detectron2's model zoo. You can use the https://dl.fbaipublicfiles... url as well
+###cfg.MODEL.WEIGHTS = model_zoo.get_checkpoint_url("COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml")
+cfg.MODEL.WEIGHTS = model_zoo.get_checkpoint_url("COCO-Detection/faster_rcnn_R_50_FPN_3x.yaml")
+predictor = DefaultPredictor(cfg)
+
+# %%
+# We use this little helper method to overlay the model predictions on a
+# given image.
+def predict_and_overlay(model, filename):
+    # helper method to run the model on an image and overlay the predictions
+    im = cv2.imread(filename)
+    out = model(im)
+    # We can use `Visualizer` to draw the predictions on the image.
+    v = Visualizer(im[:, :, ::-1], MetadataCatalog.get(cfg.DATASETS.TRAIN[0]), scale=1.2)
+    out = v.draw_instance_predictions(out["instances"].to("cpu"))
+    plt.figure(figsize=(16,12))
+    plt.imshow(out.get_image()[:, :, ::-1])
+    plt.axis('off')
+    plt.tight_layout()
+
+# %%
+# The lightly framework expects a certain bounding box and prediction format. 
+# We create another helper method to convert the detectron2 output into 
+# the desired format.
+def convert_bbox_detectron2lightly(outputs):
+    # convert detectron2 predictions into lightly format
+    height, width = outputs['instances'].image_size
+    boxes = []
+
+    for (bbox_raw, score, class_idx) in zip(outputs['instances'].pred_boxes.tensor, 
+                                            outputs['instances'].scores,
+                                            outputs['instances'].pred_classes):
+        x0, y0, x1, y1 = bbox_raw.cpu().numpy()
+        x0 /= width
+        y0 /= height
+        x1 /= width
+        y1 /= height
+      
+        boxes.append(BoundingBox(x0, y0, x1, y1))
+    output = ObjectDetectionOutput.from_scores(
+      boxes, outputs['instances'].scores.cpu().numpy(),
+      outputs['instances'].pred_classes.cpu().numpy().tolist())
+    return output
+
+# %%
+# Run model predictions
+# ----------------------
+#
+# We now use the created model and iterate over the `unlabeled_set` and make predictions.
```

Comment:
I would not use the `unlabeled_set`, but `query_set` here. I agree that `unlabeled_set` is a better name for this tutorial, but it could easily happen, that users will predict on the `unlabeled_set` also for the second iteration of AL. This will cause an error, as predictions are needed for the full `query_set`, no the `unlabeled_set`.